lang/src/gen/build.rs

```diff
@@ -48,6 +49,31 @@ pub fn generate_code(tokens: &mut TokenStream2, contract: &hir::Contract) {
     codegen_for_event_mod(tokens, contract);
 }
 
+fn codegen_for_contract_env(tokens: &mut TokenStream2, contract: &hir::Contract) {
+    let env_types = &contract.env_types_type;
+    tokens.extend(quote! {
+        mod types {
+            use super::*;
+            use ink_core::env::{ContractEnv, EnvTypes};
+
+            pub type AccountId = <ContractEnv<#env_types> as EnvTypes>::AccountId;
+            pub type Balance = <ContractEnv<#env_types> as EnvTypes>::Balance;
+            pub type Hash = <ContractEnv<#env_types> as EnvTypes>::Hash;
+            pub type Moment = <ContractEnv<#env_types> as EnvTypes>::Moment;
+        }
+
+        use types::{
+            AccountId,
+            Balance,
+            Hash,
+            Moment,
+        };
+
+        use ink_core::env::Env as _;
+        #[allow(snake_case)] type env = ink_core::env::ContractEnv<#env_types>;
```

Comment:
Let's hope that this dirty trick is not too dirty. :D

Reply:
so we could in theory use `ink_model::ExecHandler<env>` ?
Not that we should!

Reply:
I'm not 100% sold on this. I propose we stick with this for this PR. I've opened an issue to address this: #125

Reply:
Removed